src/domain/blockchain/usecase/FetchNewOracleRequestsUseCase.spec.js

```diff
@@ -0,0 +1,36 @@
+const { describe, it, beforeEach } = require('mocha');
+const { expect } = require('chai');
+const FetchNewOracleRequestsUseCase = require('./FetchNewOracleRequestsUseCase');
+const { logger } = require('../../common/utils/TestMocks');
+
+describe('FetchNewOracleRequestUseCase', () => {
+  const oracle = () => ({
+    // eslint-disable-next-line no-unused-vars
+    getRequests: (fromBlock, toBlock) => Promise.resolve([{ id: 'abc', url: 'abc', validFrom: Date.now() }]),
+  });
+
+  let sut;
+
+  beforeEach(() => {
+    // given
+    sut = new FetchNewOracleRequestsUseCase(oracle(), logger(), 1);
+  });
+
+  it('should fetch new oracle responses', async () => {
+    // when
+    const requests = await sut.fetchNewRequests(4);
+    // then
+    expect(requests).to.be.an.instanceof(Array);
+    expect(requests).to.have.lengthOf(1);
+    expect(sut.lastBlock).to.equal(4);
+    expect(sut.logger.list().length).to.equal(1);
+  });
+
+  it('should return empty array if new block is equal to current one', async () => {
+    // when
+    const requests = await sut.fetchNewRequests(0);
```

Comment:
`beforeEach` creates the use case with `lastBlock == 1` so here the new block number is not equal current block number

Reply:
Nope. You can see it in FetchNewOracleUseCase. In constructor there is a startBlockNumber parameter and it's converted into last block in the body.